File: app/src/main/java/com/kacper/krakgo/helpers/SharedPreferencesHelper.java
package com.kacper.krakgo.helpers;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

import com.kacper.krakgo.KrakGoApp;

/**
 * Created by kacper on 23/10/2017.
 */

public class SharedPreferencesHelper {
    private static final String PREFERENCES_NAME = "KrakGoShared";
    public static final String REMEMBER_USER = "remember_user";

    public static void saveToSharedPreferences(String key, boolean value){
        if(KrakGoApp.getApplicationCtx() != null) {
            SharedPreferences sharedPref = KrakGoApp.getApplicationCtx()
                    .getSharedPreferences(PREFERENCES_NAME, 0);
            SharedPreferences.Editor editor = sharedPref.edit();
            editor.putBoolean(key, value);
            editor.commit();
        }
    }
    public static Boolean getBoolean( String key, boolean defaultValue){
        SharedPreferences sharedPref = KrakGoApp.getApplicationCtx()
                .getSharedPreferences(PREFERENCES_NAME, 0);
        return sharedPref.getBoolean(key, defaultValue);
    }
    public static Boolean getBoolean(String key){
        return getBoolean(key, false);
    }
}

File: app/src/main/java/com/kacper/krakgo/helpers/SharedPreferencesHelper.kt
package com.kacper.krakgo.helpers

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences

import com.kacper.krakgo.KrakGoApp

/**
 * Created by kacper on 23/10/2017.
 */

object SharedPreferencesHelper {
    private val PREFERENCES_NAME = "KrakGoShared"
    val REMEMBER_USER = "remember_user"

    fun saveToSharedPreferences(context: Context, key: String, value: Boolean) {

        val sharedPref = context.getSharedPreferences(PREFERENCES_NAME, 0)
        val editor = sharedPref.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(context: Context, key: String, defaultValue: Boolean): Boolean {
        val sharedPref = context.getSharedPreferences(PREFERENCES_NAME, 0)
        return sharedPref.getBoolean(key, defaultValue)
    }

    fun getBoolean(key: String, context: Context): Boolean {
        return getBoolean(context, key, false)
    }
}
